Add unit tests for com.thoughtworks.xstream.io.naming.StaticNameCoder

JohnBergqvist opened the pull request:
Hi,

I've analysed your code base and noticed that `com.thoughtworks.xstream.io.naming.StaticNameCoder` is not fully tested.

I've written some tests for the functions in this class with the help of [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/products).

Hopefully these tests should help you detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other particular classes that you consider important.

joehni commented:
Hi John,

thanks for your efforts. First thing to remark is, that any source code must have the XStream copyright as file header. I can add this, but only with your explicit permission.

Regarding the test itself. Well, if it is generated, it is quite impressing. However, while it increases coverage (there were no unit test at all for this class), it lacks to cover a lot of its semantic:

- an instance if unaffected by the modification of the provided maps after instantiation
- if the 2nd parameter is null, the map is shared between nodes and attributes
- the interface declares two method pairs, calling one with the result of the other should always provide the original value

Dedicating one test for each method might be a valid strategy, but it will often not cover the complete behaviour. The tests may cover the 3rd case and find a regression, but a regression in the first two cases will currently go without any notice.

Cheers,
Jörg

JohnBergqvist commented:
@joehni 

Hi Jörg,

I've now refactored the tests to address your comments. I now have a test for each of the 3 states in the constructor, while i'm now using the `encode` & `decode` methods to assert the constructor's behaviour.

I've also added a copyright header.

Let me know what you think.

Regards, 

John.

joehni commented:
Hi John, I've applied this with some modifications. Thanks for your contribution.
